readthedocs/core/resolver.py

```diff
@@ -151,27 +151,31 @@ def resolve(self, project, protocol='http', filename='', private=None,
                                    **kwargs),
         )
 
-    def _get_canonical_project(self, project):
+    def _get_canonical_project(self, project, projects=None):
         """
         Recursively get canonical project for subproject or translations.
 
         We need to recursively search here as a nested translations inside
         subprojects, and vice versa, are supported.
 
         :type project: Project
+        :type projects: List of projects for iteration
         :rtype: Project
         """
+        # Track what projects have already been traversed to avoid infinite
+        # recursion. We can't determine a root project well here, so you get
+        # what you get if you have configured your project in a strange manner
+        if projects is None:
+            projects = [project]
```

Comment:
Don't you want an `else` here that does `projects.append(project)`?

Reply:
oops, yes, that is the missing piece